### DISCUSSION TITLE
[@types/react-native] ranges in Animated.InterpolationConfigType should be readonly

### DISCUSSION THREAD

**Discussion by YasminBeatriz** [DefinitelyTyped/DefinitelyTyped#58842] [Issues with a @types package]
It's common to declare an interpolation config as readonly constants when working with Animated. However, the `inputRange` and `outputRange` arrays are [currently declared](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-native/index.d.ts#L8685) as mutable `number[]`.

So, let's say I have this config in my application:
```js
const Constants = {
  myInterpolationConfig: {
      extrapolate: 'clamp',
      inputRange: [0, 1],
      outputRange: [1, 0],
  },
} as const
```

This throws:
```ts
Argument of type '{ readonly extrapolate: "clamp"; readonly inputRange: readonly [0, 1]; readonly outputRange: readonly [1, 0]; }' is not assignable to parameter of type 'InterpolationConfigType'.
  Types of property 'inputRange' are incompatible.
    The type 'readonly [0, 1]' is 'readonly' and cannot be assigned to the mutable type 'number[]'.

57                 Constants.myInterpolationConfig,
```

This has already been addressed [here](https://github.com/facebook/react-native/commit/6584304c100ce4d51a5c4d606170a6ad0dc00875) for FlowType. Ideally not only the whole object should become readonly, but the `inputRange` and `outputRange` individually as well.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
I think most props are mutable so the current type error is probably incorrect. If this is already changed in the public flow types, I don't see a reason to not port this to the TypeScript types. Would you be willing to file a PR to fix this issue?
